graph/graph_windows.go

```diff
@@ -155,6 +182,50 @@ func (graph *Graph) TarLayer(img *Image) (arch archive.Archive, err error) {
 		// We keep this functionality here so that we can still work with the VFS
 		// driver during development. VFS is not supported (and just will not work)
 		// for Windows containers.
-		return graph.driver.Diff(img.ID, img.Parent)
+		root := graph.imageRoot(img.ID)
+		mFileName := filepath.Join(root, tarDataFileName)
+		mf, err := os.Open(mFileName)
```

Comment:
I might not have looked closely enough, but at first glance it looks like added code blocs in `graph_unix.go` and `graph_windows.go` are very similar. Could we just put the generic implementation in one separate function and have the unix and windows implementations call out to it?